compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -670,10 +679,15 @@ static void PrintService(const ServiceDescriptor* service,
   (*vars)["service_name"] = service->name();
   (*vars)["file_name"] = service->file()->name();
   (*vars)["service_class_name"] = ServiceClassName(service);
+  #ifdef GRPC_VERSION
+    (*vars)["grpc_version"] = "(version "+string(XSTR(GRPC_VERSION))+")";
+  #else
+    (*vars)["grpc_version"] = "";
+  #endif
   p->Print(
       *vars,
       "@$Generated$(\n"
-      "    value = \"by gRPC proto compiler\",\n"
+      "    value = \"by gRPC proto compiler $grpc_version$\",\n"
```

Comment:
Hah, I originally did that but reverted in favour _slightly_ nicer code. Nothing will hide from you Eric :)

Addressed.